lm/srilm.py

```diff
@@ -94,6 +95,74 @@ def hash(cls, kwargs):
         return super().hash(kwargs)
 
 
+class DiscountNgramsJob(Job):
+    """
+    Create a file with the discounted ngrams with SRILM
+    """
+
+    def __init__(
+        self,
+        ngram_order: int,
+        counts: tk.Path,
+        vocab: tk.Path,
+        count_exe: tk.Path,
+        *,
+        extra_discount_args: Optional[List[str]] = None,
+        cpu_rqmt: int = 1,
+        mem_rqmt: int = 48,
+        time_rqmt: float = 24,
+        fs_rqmt: str = "100G",
+    ):
+        self.ngram_order = ngram_order
+        self.counts = counts
+        self.vocab = vocab
+        self.count_exe = count_exe
+        self.discount_args = extra_discount_args
```

Comment:
cf line 146
```
>>> f"  {' '.join(None)} -memuse\n"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only join an iterable
```
therefore 
```suggestion
        self.discount_args = extra_discount_args or []
```

This might in line 146 add an extra space if there are no extra args, but that would only be cosmetical in a file that nobody reads anyway..